cli/src/pcluster/config/cluster_config.py

```diff
@@ -1574,6 +1576,22 @@ def _register_validators(self):
             resource_name="SlurmQueues",
         )
 
+    def get_queues_security_group_overwrite_status(self):
```

Comment:
This function is validator specific, we should move it in the validator class.

When registering the validator we can pass: `self.head_node.networking.security_groups` and 
 `self.scheduling.queues` then it's the validator that should know the logic to identify the status analyzing all the queues.

By moving the method you don't need to import the new Enum here.

Reply:
Then after moving will be easier to create unit tests for `get_queues_security_group_overwrite_status`private method of the validator (that is not yet covered).

See for example unit test for `_is_instance_type_format`. private method of `AwsBatchInstancesArchitectureCompatibilityValidator`

Reply:
We have been passing simple data types to validators, which helps to decrease the complexity of unit tests.
I agree that moving the method to the validator do reduce the complexity of source code. But it increases complexity of the unit tests, because we are passing the `queues` list of object to the validator

Reply:
By looking at our validators I can see some of them are doing a cycle for all the compute resources of a given queue. This is a similar case, here the responsibility to check the status of this parameter should be of the validator, the overwrite status is useless in the config module and we also have an Enum that is very specific to the validator so I'd move this code.

In general let's focus on source code complexity rather than unit tests complexity (even if hard to write unit tests means bad smell in the code structure).

Anyway let's talk about this before proceeding.

Reply:
Done